Update PostgreSQL password key for the official k8s chart

kooba opened the pull request:
Fixes #15

kooba committed:
Update PostgreSQL password key for the official k8s chart

Signed-off-by: Jakub Borys <jakub.borys@gmail.com>

tyler46 commented:
@kooba I think we also need to update the `README` section where the deployment of Postgresql instance is described. Something like that will do the job:

`$ helm --kube-context=minikube install --name db --namespace examples stable/postgresql --set postgresqlDatabase=orders`

What do you think ?

kooba commented:
Good catch @tyler46 , I'll have to do more testing to verify nothing else has changed in official charts.

tyler46 commented:
@kooba I think that redis helm chart  has also changed. By default redis instance has master - slave topology enabled. Because of this, master redis is named as `cache-redis-master`. Disabling cluster topology will still mark the single redis instance as `cache-redis-master`, so it doesn't fix the issue as initially I have thought. Thus `products` service cannot connect to redis, since the `REDIS_HOST` is different than the one defined in `charts/products/templates/deployment.yaml`. To fix it, we can update the following section on `deployments.yaml`:
```
env:
   - name: REDIS_HOST
     value: cache-redis-master
```

kooba committed:
Fix to redis deployment. Makefile improvements.

kooba committed:
Examples are fixed

tyler46 commented:
thank you @kooba :+1:

kooba committed:
Move examples to use docker-desktop k8s cluster

kooba commented:
@tyler46 the issues with Redis and Posgresql chart changes are now addressed. Also moved to use `docker-for-desktop` as k8s cluster for the readme as it's much simpler to get going with...

mattbennett commented on k8s/README.md:
```diff
@@ -2,56 +2,32 @@
-* [VirtualBox](https://www.virtualbox.org/wiki/Downloads)
-* [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
-* [Minikube](https://github.com/kubernetes/minikube#installation)
+* [docker-for-desktop](https://docs.docker.com/docker-for-mac/) with Kubernetes enabled
```
Is there a non Mac-specific version of this page?

kooba replied:
Not really, but I can point to both [mac](https://docs.docker.com/docker-for-mac/) and [windows](https://docs.docker.com/docker-for-windows/)? Although no way I can test this on Windows at the moment.

mattbennett replied:
Perhaps just add a comment saying we assume Mac for simplicity but it should work on Windows or indeed any other Kubernetes cluster.

mattbennett commented on k8s/README.md:
````diff
@@ -215,15 +182,19 @@ As you can see this template is using values coming from `Chart` and `Values` fi
 
-`$ minikube addons enable ingress`
+```sh
+$ kubectl --context=docker-desktop apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/master/deploy/static/mandatory.yaml
````
Does the chart installation of nginx-ingress not work on docker-desktop clusters?

kooba replied:
Not sure what you mean. It is installed to `docker-desktop` cluster here.

kooba replied:
I think maybe you're asking if it would work on docker-desktop clusters... yes it works fine: https://kubernetes.github.io/ingress-nginx/deploy/#docker-for-mac ... the only restriction is that it will only listen on port 80 or 443. I think I could further simplify this tutorial by forgoing ingress altogether and setting gateway`service` definition to be of `NodePort` type which would allow to expose that service on any port we desire... which maybe would play better for people with existing `docker-desktop` clusters.

mattbennett replied:
I mean why are you installing manifest directly rather than `chart upgrade --install nginx-ingress`?

kooba replied:
Just tested it with ingress controller installed via helm chart and doesn't seem to play well with docker for desktop for some reason... It forces redirection to https 🤔

mattbennett replied:
This was asked mostly out of curiosity. Feel free to merge!

mattbennett reviewed (approved):
Looks great. Just a couple of small comments

kooba committed:
Comment on other clusters

kooba committed:
Using Helm v3

Signed-off-by: Jakub Borys <jakub.borys@gmail.com>

kooba commented:
@mattbennett having trouble getting ingress controller installed via helm chart working. Would like to merge this.
